content/blog/intro-to-mdox/index.md

```diff
@@ -0,0 +1,197 @@
+
+## mdox: All in one solution and more!
+
+Fortunately, there is now a way to tackle all these problems (and more) automatically using `mdox`! Great tooling should get out of your way and let you do exactly what you need to without causing pain, which is what we want to achieve with `mdox`.
+
+[`mdox`](https://github.com/bwplotka/mdox) is a simple Go CLI tool, you can install using `go get` or pin using [bingo](https://github.com/bwplotka/bingo).
+
+### Auto-formatting
+
+Formatted markdown allows writers to understand documentation quickly and avoids surprises. On the other hand, it’s important to not expect writers to write perfect, formatted markdowns in a way you want. Even if you check certain formatting on CI, writers will get quickly annoyed when thrown into the loop of pushing changes, waiting for CI linter, see an error (output may just be the first error or simply incomprehensible), push again, learn about a new error elsewhere and repeat.. and lose a few hours in the process.
+
+`mdox` solves those issues by formatting markdown files automatically, following the GitHub Flavored Markdown spec, and fails if it cannot format them. Simply, run `mdox fmt <files>`! If you want to run in CI, use `mdox fmt --check <files>`, which does not modify anything, just checks the output produced by `mdox` would be different than what was committed. If not, it will fail and specify exact diffs of what is not formatted or wrong.
+
+![mdox showing diff on error](mdox_check.png)
+
+`fmt` command ensures a smooth automatic experience for the contributor that also does not compromise the markdown quality of the project. Win-win situation!
+
+> Kudos to some Go developers who passionately started `[Kunde21/markdownfmt](https://github.com/Kunde21/markdownfmt)` fork, our tool used internally for formatting, with [goldmark](https://github.com/yuin/goldmark) parser help You might be surprised, but parsing and formatting markdown is quite advanced engineering in itself! We now maintain it, together with [https://github.com/karelbilek](https://github.com/karelbilek), [https://github.com/Kunde21](https://github.com/Kunde21) - a great open-source example!
+
+### Link Checking
+
+Broken links are super annoying, especially in software documentation. Trying to figure out the correct link as you're trying to learn something else is a distracting context switch. We can mitigate this by using a markdown link checker. However, most currently available link checkers seem to be underperforming, in a few ways:
+
+* They just keep on visiting links. This results in obvious rate-limiting ("Too many requests" errors) especially from very often linked resources like GitHub (Just imagine CHANGELOG (: ). You can fully disable checking any GitHub link, but then might have surprises later on! You can enable retries or dev APIs in those cases, but still, it would either take a huge time or will be hard to set up.
```

Comment:
```suggestion
* They just keep on visiting links. This results in obvious rate-limiting ("Too many requests" errors) especially from very often linked resources like GitHub (Just imagine CHANGELOG (: ). You can fully disable checking such links, but then might have surprises later on! You can enable retries or dev APIs in those cases, but still, it would either take a huge time or will be hard to set up.
```

GitHub repeats a lot